mesonbuild/compilers/compilers.py

```diff
@@ -697,21 +719,13 @@ def cached_compile(self, code, cdata: coredata.CoreData, *, extra_args=None, mod
         # Check if not cached
         if key not in cdata.compiler_check_cache:
             with self.compile(code, extra_args=extra_args, mode=mode, want_output=False, temp_dir=temp_dir) as p:
-                # Remove all attributes except the following
-                # This way the object can be serialized
-                tokeep = ['args', 'commands', 'input_name', 'output_name',
-                          'pid', 'returncode', 'stdo', 'stde', 'text_mode']
-                todel = [x for x in vars(p).keys() if x not in tokeep]
-                for i in todel:
-                    delattr(p, i)
-                p.cached = False
                 cdata.compiler_check_cache[key] = p
                 yield p
+                p.cached = True
```

Comment:
This change is incorrect, but I see that you've fixed it in [a later commit](https://github.com/mesonbuild/meson/pull/7759/commits/e4fb6d02d9aa418296033ff2bb587c21854fccbe) so we can let it be.

Reply:
This isn't incorrect per se, but its entirely too clever, which is why I dropped it.